### PR TITLE
feat: install extra deps for ember project

### PR DESCRIPTION
**Pull request by haochuan**
PR for #463, CLI would hard stop and show the error before if running migration on an Ember project that doesn't meet the requirements:

```
rehearsal migrate
✖ Your Ember project does not meet the requirements of migration. Please make sure you have:
  - ✅ Project Node Version v16.18.1 should >= 16.0.0
  - ❌ Dependency ember-source ^3.10.0 should >=3.24.0
  - ❌ Dependency ember-modifier 3.1.7 should >=3.2.7
◼ Initialization
◼ Installing dependencies
◼ Creating tsconfig.json
◼ Converting JS files to TS
◼ Creating eslint config
```

Also `exitOnError` has been changed to `true` in `rehearsal migrate`, in order to hard stop the CLI if there is any issue. The reason we had it on `false` was that we don't want to quit the CLI if there we got non 0 exit during `tsc` check. This would also cover #481.

### CODE REVIEW

File: packages/cli/src/commands/migrate.ts [hunk 1]
```diff
@@ -182,13 +183,32 @@ migrateCommand
...
               await _ctx.userConfig.install();
             }
+
             // even if typescript is installed, exec this and get the latest patch
             await addDep(['typescript'], true, { cwd: options.basePath });
+
+            // extra dependencies for Ember App/Addon/Engine
+            // TODO: dependes on how much extra stuff we need for a specific framework,
+            // probably need a plugable system for this.
+            if (isEmber(packageJson)) {
```
Comment (lynchbomb) [inline packages/cli/src/commands/migrate.ts:199]: we will need a conditional around which version of Ember and which version of Node. 

Glint only supports:
```
Ember 3.24+
Node 16+
```

**Review by lynchbomb — CHANGES_REQUESTED** [review]

File: packages/cli/src/commands/migrate.ts (hunk 1 above)
Comment (lynchbomb) [reply packages/cli/src/commands/migrate.ts:199]: ... and 

`ember-modifier 3.2.7+`

Comment (egorio) [reply packages/cli/src/commands/migrate.ts:199]: Weird to see this as a part of migrate.ts. Isn't better to move this part to a framework-specific package/function (like we did with migration --grapth / migration-graph-ember)?

Move this `['@glint/core', '@glint/template', '@glint/environment-ember-loose']` outside of migrate.ts (params?) and just check `if (additionalDependencies) { ... install them ...}`?

Comment (haochuan) [reply packages/cli/src/commands/migrate.ts:199]: @egorio yeah totally agree, that's why in the comment I have the TODO to take it out of the main cli and make it like a plugin-ish structure, still deciding where those should be and I'm open to any suggestions 😄

Comment (egorio) [reply packages/cli/src/commands/migrate.ts:199]: > probably just move it to a new file

Oh, didn't read a comment :) I think it's fine to put in a separate file for now :)

**Comment by haochuan** [comment]
Having a wired issue during `./node_modules/.bin/ember install ember-cli-typescipt@latest`, which failed the test:

```
Command failed with exit code 1: npm install --save-dev --loglevel error ember-cli-typescript@latest
npm ERR! Cannot read properties of null (reading 'name')
```

**Comment by lynchbomb** [comment]
@haochuan i've not debugged that comment above. My knee-jerk is `npm install` <<< if this is trying to run on this repo we are using `pnpm` and `npm` execs will throw.

**Comment by haochuan** [comment]
> @haochuan i've not debugged that comment above. My knee-jerk is `npm install` <<< if this is trying to run on this repo we are using `pnpm` and `npm` execs will throw.

Yeah seems like all the utils function to detect 'moduleManager` etc would get the value from `process.cwd()` instead of the actual test tmp dir, I'll update them to see if that would solve the issue

**Comment by haochuan** [comment]
Synced with @lynchbomb  and we both agreed to take any framework specific stuff out of the main CLI. I'm going to close this PR and try to see what I can do to have this ember validator and extra deps installer using the existed `userConfig`

**Pull request closed without merging**